biz.aQute.bndlib/src/aQute/bnd/osgi/Verifier.java

```diff
@@ -531,6 +532,24 @@ public void verify() throws Exception {
 		verifyCapabilities();
 		verifyMetaPersistence();
 		verifyPathNames();
+
+		doVerifierPlugins();
+	}
+
+	private void doVerifierPlugins() {
+		for (VerifierPlugin plugin : getPlugins(VerifierPlugin.class)) {
+			Processor previous = beginHandleErrors(plugin.toString());
+			try {
+				plugin.verify(analyzer);
+			}
+			catch (Exception e) {
+				e.printStackTrace(System.err);
+				error("Verifier Plugin %s failed %s", plugin, e);
+			}
+			finally {
+				endHandleErrors(previous);
```

Comment:
This is wrong too. The error call in the exception block will be be in the wrong Processor. You need to wrap the try/finally just around the call to plugin.verify.